easybuild/easyblocks/i/impi.py

```diff
@@ -126,7 +126,8 @@ def install_step(self):
             run_cmd(cmd, log_all=True, simple=True)
 
         # recompile libfabric (if requested)
-        if impiver >= LooseVersion('2019') and self.cfg['libfabric_rebuild']:
+        if (impiver >= LooseVersion('2019') and self.cfg['libfabric_rebuild'] and
+                os.path.exists(os.path.join(self.installdir, 'libfabric'))):
             if self.cfg['ofi_internal']:
                 change_dir(os.path.join(self.installdir, 'libfabric'))
```

Comment:
@bartoldeman Let's introduce an additional variable for the `libfabric` path?

```python
if impiver >= LooseVersion('2019') and self.cfg['libfabric_rebuild']:
    libfabric_path = os.path.join(self.installdir, 'libfabric')
    # some Intel MPI versions (like 2019 update 5) no longer ship libfabric sources
    if self.cfg['ofi_internal'] and os.path.exists(libfabric_path):
        change_dir(libfabric_path)

Reply:
I can't move the condition to the second `if` statement (the `else` won't work properly then), but I could introduce the variable so I did that. Also the comment applies to update 6 only (I checked update 5!)